includes/functions.php

```diff
@@ -103,11 +115,12 @@ function get_site_terms ($blog_id = 0) {
      *
      * @param WP_Term $term
      * @param array $args
+     * @param array $exclude Optional. IDs of categories to exclude. Sites with these categories will be excluded from the list.
      *
      * @return array
      */
-    function get_sites_in_directory_by_term ($term, $args = array()) {
-        $args = wp_parse_args($args, array(
+    function get_sites_in_directory_by_term ($term, $args = array(), $exclude = array() ) {
```

Comment:
This one wasn't so simple, because `get_sites_in_directory_by_term()` already accepts an array of arguments, plus I need two additional `'tax_query'` parameters (on top of that used for the basic query) for the exclude functionality to work. I added a `json_decode()` to convert the parameters passed via the `'query_args'` shortcode attribute and used `array_merge_recursive` to combine the different arrays of parameters. It looks like a lot more work this way, but the function call is cleaner now :)

I also added support for `'orderby'` and `'order'` shortcode attributes. However, if `'orderby'` is `'blogname'` the sort will be done using the directory entries' titles which, AFAICT, are not updated when the site title is changed. So I have an additional check to (re)sort the `$details` array after `get_posts()` in that case.